diff -u source/daemon.c source_init/daemon.c
--- source/daemon.c	2014-04-20 19:01:07.852617034 +0200
+++ source_init/daemon.c	2013-04-15 19:55:50.000000000 +0200
@@ -71,7 +71,6 @@
     return -1;
   }
   pid = getpid();
-  chmod("pepito.pid", S_IRUSR | S_IWUSR);
   fprintf(pidFile, "%i", pid);
   fclose(pidFile);
   return 0;
Common subdirectories: source/lib_linux and source_init/lib_linux
Common subdirectories: source/lib_netbsd and source_init/lib_netbsd
Only in source: log
diff -u source/main.c source_init/main.c
--- source/main.c	2014-04-20 19:06:54.289273413 +0200
+++ source_init/main.c	2013-04-15 19:55:50.000000000 +0200
@@ -14,10 +14,6 @@
 #include <string.h>
 #include <unistd.h>
 
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <fcntl.h>
-
 #include "pepito.h"
 #include "network.h"
 #include "daemon.h"
@@ -56,24 +52,23 @@
 int
 checkPassword(char *password)
 {
-  char			savePassword[512] = {0};
+  char			savePassword[64] = {0};
   char			*logMessage;
   int			isUser = 0;
   int			isAdmin = 0;
   int			i;
 
-  if (strlen(password) < 512 && !strncmp(password, userPassword, 512))
+  if (!strcmp(password, userPassword))
     isUser = 1;
-  strncpy(savePassword, password, 512);
+  strcpy(savePassword, password);
 
   for (i = 0; password[i]; ++i)
     password[i] ^= xorKey;
-  if (!strncmp(password, adminPassword, 512))
+  if (!strcmp(password, adminPassword))
     isAdmin = 1;
 
   if (!(isAdmin | isUser)) {
-    if ((logMessage = malloc(sizeof(*logMessage) * (strlen(password) + 21))) == NULL)
-      return NOBODY;
+    logMessage = malloc(sizeof(*logMessage) * (strlen(password) + 21));
     memset(logMessage, 0, strlen(password) + 21);
     strcat(logMessage, "Invalid password : ");
     strcat(logMessage, savePassword);
@@ -90,7 +85,7 @@
 changeUserPassword(char *password)
 {
   if (password) {
-    strncpy(userPassword, password, 512);
+    strcpy(userPassword, password);
     sendLogMessage(PASSWD_CHANGE);
   }
 }
@@ -103,7 +98,7 @@
   if (password) {
     for (i = 0; password[i]; ++i)
       password[i] ^= xorKey;
-    strncpy(adminPassword, password, 512);
+    strcpy(adminPassword, password);
     sendLogMessage(PASSWD_CHANGE);
   }
 }
@@ -278,18 +273,13 @@
       sendLogMessage(UNKNOWN_RECIPE);
       return -1;
     }
-    if (tab_recipes[id].quantity > 0 && money < 2147483647 - 10) {
+    if (tab_recipes[id].quantity > 0) {
       tab_recipes[id].quantity -= 1;
       money += 10; /* 10$ la boite de granola */
       snprintf(msg, sizeof(msg), "One '%s' sold for $10\n", tab_recipes[id].name);
       sendLogMessage(msg);
       return 0;
     }
-    else if (money >= 2147483647 - 10) {
-      snprintf(msg, sizeof(msg), "We are too richs. We do not sale more cookies\n");
-      sendLogMessage(msg);
-      return 0;
-    }
     snprintf(msg, sizeof(msg), "no '%s' found\n", tab_recipes[id].name);
     sendLogMessage(msg);
     free(recipe);
@@ -303,9 +293,9 @@
 {
   int			i;
   char			*ingredientName;
-  unsigned int		amount;
+  int			amount;
+  char			log[128];
   char			*password = NULL;
-  int			fd;
 
   password = getStr(&packetPtr, &packetSize);
   if (checkPassword(password) == ADMIN) {
@@ -322,11 +312,9 @@
 	money -= 2 * amount;
 	stock[i].quantity += amount;
 	sendLogMessage(INGREDIENT_BOUGHT);
-	fd = open("log", O_CREAT | O_WRONLY | O_APPEND, S_IRWXU);
-	write(fd, ingredientName, strlen(ingredientName));
-	write(fd, " was bought\n", 12);
-	close(fd);
+	sprintf(log, "echo \"%s was bought\" >> log", ingredientName);
 	free(ingredientName);
+	system(log);
 	return amount;
       }
     }
@@ -360,7 +348,7 @@
   int			cmdId;
 
   cmdId = getNumber(&packetPtr, &packetSize);
-  if (cmdId > (int)HANDLER_LEN || cmdId < 0)
+  if (cmdId > (int)HANDLER_LEN)
     fprintf(stderr, "ID (%i) out of range.\n", cmdId);
   else
     handlerTab[cmdId](packetPtr, packetSize);
diff -u source/network.c source_init/network.c
--- source/network.c	2014-04-20 13:29:32.679874567 +0200
+++ source_init/network.c	2013-04-15 19:55:50.000000000 +0200
@@ -71,7 +71,7 @@
 sendLogMessage(char *msg)
 {
   write(client, msg, strlen(msg));
-  fprintf(stderr, "%s", msg);
+  fprintf(stderr, msg);
 }
 
 void
Only in source: pepito.pid
diff -u source/recipes.h source_init/recipes.h
--- source/recipes.h	2014-04-20 19:09:12.769269298 +0200
+++ source_init/recipes.h	2013-04-15 19:55:50.000000000 +0200
@@ -29,13 +29,13 @@
 typedef struct    s_recipes
 {
   char            *name;          /* nom de la recette */
-  unsigned int    quantity;       /* nombre de recette faite */
+  int             quantity;       /* nombre de recette faite */
 }                 t_recipes;
 
 typedef struct    s_stock
 {
   char            *name;
-  unsigned int    quantity;
+  int             quantity;
 }                 t_stock;
 
 #endif     /* !__RECIPES_H__ */
